scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/Worker.java

```diff
@@ -89,9 +97,29 @@ private static TaskState getInitialStateForStreams(Map<TaskId, SortedSet<StreamI
      * This includes fetching saved state of each task or creating a new initial
      * state for tasks that haven't run successfully before.
      */
-    private TaskActionsQueue queueFirstActionForEachTask(Map<TaskId, SortedSet<StreamId>> groupedStreams) throws ExecutionException, InterruptedException {
-        return new TaskActionsQueue(createTasksWithState(groupedStreams)
-                .map(task -> TaskAction.createFirstAction(workerConfiguration, task)).collect(Collectors.toSet()));
+    private Collection<TaskAction> queueFirstActionForEachTask(Map<TaskId, SortedSet<StreamId>> groupedStreams)
+            throws ExecutionException, InterruptedException {
+        return createTasksWithState(groupedStreams).map(task -> TaskAction.createFirstAction(workerConfiguration, task))
+                .collect(Collectors.toSet());
+    }
+
+    private boolean shouldStop() {
+        return shouldStop;
+    }
+
+    private ScheduledExecutorService getExecutorService() {
+        return workerConfiguration.getExecutorService();
+    }
+
+    private Callable<Object> makeCallable(TaskAction a) {
```

Comment:
Why Callable not Runnable when we're not interested in any returned value?

Reply:
Exceptions.